hydra-cardano-api/src/Hydra/Cardano/Api/PolicyId.hs

```diff
@@ -2,7 +2,24 @@ module Hydra.Cardano.Api.PolicyId where
 
 import Hydra.Cardano.Api.Prelude
 
+import Cardano.Ledger.Alonzo.TxInfo qualified as Ledger
 import Cardano.Ledger.Hashes qualified as Ledger
+import Cardano.Ledger.Mary.Value qualified as Ledger
+import Hydra.Cardano.Api.Hash (unsafeScriptHashFromBytes)
+import PlutusLedgerApi.V2 (CurrencySymbol, fromBuiltin, unCurrencySymbol)
 
 toLedgerScriptHash :: PolicyId -> Ledger.ScriptHash StandardCrypto
 toLedgerScriptHash (PolicyId scriptHash) = toShelleyScriptHash scriptHash
+
+-- | Convert Cardano api 'PolicyId' to Cardano ledger `PolicyID`.
+toLedgerPolicyID :: PolicyId -> Ledger.PolicyID StandardCrypto
+toLedgerPolicyID (PolicyId sh) = Ledger.PolicyID (toShelleyScriptHash sh)
+
+-- | Convert Cardano api 'PolicyId' to Plutus `CurrencySymbol`.
+toPlutusCurrencySymbol :: PolicyId -> CurrencySymbol
+toPlutusCurrencySymbol = Ledger.transPolicyID . toLedgerPolicyID
+
+-- | Convert a plutus 'CurrencySymbol' into a cardano-api 'PolicyId'.
+fromPlutusCurrencySymbol :: CurrencySymbol -> PolicyId
+fromPlutusCurrencySymbol = PolicyId . unsafeScriptHashFromBytes . fromBuiltin . unCurrencySymbol
+{-# DEPRECATED fromPlutusCurrencySymbol "this is partial" #-}
```

Comment:
tried to address this but found my self unsure about how to handle this function when calling it from `headIdToPolicyId`, as this is being used to prepare all our txs in `module Hydra.Chain.Direct.State`.

from my understanding we need to pass an error handler as part of `headIdToPolicyId` and then treat it properly on our tx builders so that in case this functions throws we transform it to the proper TxError.

wdyt?

shall we address this as part of this PR or should be leftover for later?